web-common/src/features/dashboards/filters/dimension-filters/DimensionFilter.svelte

```diff
@@ -113,29 +115,87 @@
     </Tooltip>
   </DropdownMenu.Trigger>
 
-  <SearchableMenuContent
-    {onSelect}
-    {onToggleSelectAll}
-    bind:searchText
-    showXForSelected={excludeMode}
-    selectedItems={[selectedValues]}
-    allowMultiSelect={true}
-    selectableGroups={[
-      {
-        name: "DIMENSIONS",
-        items: $searchValues.map((dimensionValue) => ({
-          name: dimensionValue,
-          label: dimensionValue,
-        })),
-      },
-    ]}
+  <!-- There will be some custom controls for this. Until we have the full design have a custom dropdown here. -->
```

Comment:
I'm a little worried about divergence between our menu components. For example, all other `SearchableMenuContent` would also benefit from the loading & error states you've added here. I wonder if we could add the requisite functionality to `SearchableMenuComponent` (e.g. make it pluggable?). I'll leave it up to you.

Reply:
Ya I wanted to wait for the design to stabilise before refactoring this too much. The end goal would be a common component with hooks for dimension filter.